In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from scipy import stats

# Loading the Dataset

In [2]:
salesdata = pd.read_csv("Sales_add.csv")
salesdata.head()

Month      Region      Manager  Sales_before_digital_add(in $)  \
0  Month-1  Region - A  Manager - A                          132921   
1  Month-2  Region - A  Manager - C                          149559   
2  Month-3  Region - B  Manager - A                          146278   
3  Month-4  Region - B  Manager - B                          152167   
4  Month-5  Region - C  Manager - B                          159525   

   Sales_After_digital_add(in $)  
0                         270390  
1                         223334  
2                         244243  
3                         231808  
4                         258402

# Checking the null values

In [4]:
salesdata.isnull().sum()

Month                             0
Region                            0
Manager                           0
Sales_before_digital_add(in $)    0
Sales_After_digital_add(in $)     0
dtype: int64

In [7]:
salesdata.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22 entries, 0 to 21
Data columns (total 5 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Month                           22 non-null     object
 1   Region                          22 non-null     object
 2   Manager                         22 non-null     object
 3   Sales_before_digital_add(in $)  22 non-null     int64 
 4   Sales_After_digital_add(in $)   22 non-null     int64 
dtypes: int64(2), object(3)
memory usage: 1008.0+ bytes


In [9]:
salesdata.describe()

Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
count                       22.000000                      22.000000
mean                    149239.954545                  231123.727273
std                      14844.042921                   25556.777061
min                     130263.000000                  187305.000000
25%                     138087.750000                  214960.750000
50%                     147444.000000                  229986.500000
75%                     157627.500000                  250909.000000
max                     178939.000000                  276279.000000

- There are 22 rows and 5 features in the dataset.
- There are no missing values in the data set.

# 1. Whether there is any increase in sales after stepping into digital marketing.

Hypothesis

- H0: no increase in sales after digital marketing
- H1: increase in sales after digital marketing

In [13]:
stats.ttest_rel(salesdata['Sales_After_digital_add(in $)'], salesdata['Sales_before_digital_add(in $)'],alternative='greater')

Ttest_relResult(statistic=12.09070525287017, pvalue=3.168333502287889e-11)

t statistic value is 12.091 and p value is 0.

As p value is less than 0.05, H0 is rejected. 
So there is an increase in the sales after the digital marketing

# 2. Testing dependency between the features “Region” and “Manager”.


- H0: There is no dependancy between Region & Manager
- H1: There is dependancy between Region & Manager

In [14]:
salesdata['Region'].value_counts() 

Region - A    10
Region - B     7
Region - C     5
Name: Region, dtype: int64

In [15]:
salesdata['Manager'].value_counts() 

Manager - A    9
Manager - B    7
Manager - C    6
Name: Manager, dtype: int64

In [19]:
crossTab=pd.crosstab(salesdata.Region,salesdata.Manager) # frequencies of observation
crossTab

Manager     Manager - A  Manager - B  Manager - C
Region                                           
Region - A            4            3            3
Region - B            4            1            2
Region - C            1            3            1

In [21]:
stats.chi2_contingency(crossTab)

(3.050566893424036,
 0.5493991051158094,
 4,
 array([[4.09090909, 3.18181818, 2.72727273],
        [2.86363636, 2.22727273, 1.90909091],
        [2.04545455, 1.59090909, 1.36363636]]))

chisquare_value = 3.0506 p_value = 0.5494, degrees of freedom = 4. Array gives the frequencies expected

Here p-value > 0.05 this means that we take the null hypothesis, So there is no depandancy between the Region & Manager